src/module/actor/character/feats.ts

```diff
@@ -42,15 +43,27 @@ class CharacterFeats extends Collection<FeatCategory> {
         this.createCategory({
             id: "ancestry",
             label: "PF2E.FeatAncestryHeader",
-            featFilter: "ancestry-" + actor.ancestry?.slug,
+            featFilter: actor.system.details.ancestry?.trait
+                ? `ancestry-${actor.system.details.ancestry?.trait}`
+                : null,
             supported: ["ancestry"],
             levels: classFeatSlots?.ancestry ?? [],
         });
+
+        // Attempt to acquire the trait corresponding with actor's class, falling back to homebrew variations
+        const classSlug = actor.class ? actor.class.slug ?? sluggify(actor.class.name) : null;
+        const classTrait =
+            (classSlug ?? "") in CONFIG.PF2E.featTraits
+                ? classSlug
+                : `hb_${classSlug}` in CONFIG.PF2E.featTraits
+                ? `hb_${classSlug}`
+                : null;
+
         this.createCategory({
             id: "class",
             label: "PF2E.FeatClassHeader",
-            featFilter: "classes-" + actor.class?.slug,
-            supported: ["class"],
+            featFilter: classTrait ? `traits-${classTrait},traits-archetype` : null,
+            supported: ["class", "archetype"],
```

Comment:
Interestingly enough, this doesn't do anything. archetype resolves into either skill or class depending on whether it has the skill trait, according to isFeatValid(). You can change that behavior if you want, but we'd end up allowing archetype skill feats in class/archetype feat slots.

Reply:
archetype skill feats are of the "skill" feat type. Incidentally, we don't really need to have a feat type for anything except features.

Reply:
archetype class feats are of the "class" feat type :)

Reply:
naw

Reply:
cf.,
- https://github.com/foundryvtt/pf2e/blob/master/packs/data/feats.db/sentinel-dedication.json
- https://github.com/foundryvtt/pf2e/blob/master/packs/data/feats.db/steel-skin.json